docs/source/configuration/config_loader_migration.md

```diff
@@ -60,3 +60,177 @@ In this example, `"catalog"` is the key to the default catalog patterns specifie
+
+Suppose you are migrating a templated **catalog** file from using `TemplatedConfigLoader` to `OmegaConfigLoader` you would do the following:
+1. Rename `conf/base/globals.yml` to match the patterns specified for catalog (`["catalog*", "catalog*/**", "**/catalog*"]`), e.g. `conf/base/catalog_globals.yml`
+2. Add an underscore `_` to any catalog template values. This is needed because of how catalog entries are validated.
+
+```diff
+- bucket_name: "my_s3_bucket"
++ _bucket_name: "my_s3_bucket" # kedro requires `_` to mark templatable keys
+- key_prefix: "my/key/prefix/"
++ _key_prefix: "my/key/prefix/"
+
+- datasets:
++ _datasets:
+    csv: "pandas.CSVDataSet"
+    spark: "spark.SparkDataSet"
+
+```
+
+3. Update `catalog.yml` with the underscores `_` at the beginning of the templated value names.
+```diff
+raw_boat_data:
+-   type: "${datasets.spark}"
++   type: "${_datasets.spark}"
+-   filepath: "s3a://${bucket_name}/${key_prefix}/raw/boats.csv"
++   filepath: "s3a://${_bucket_name}/${_key_prefix}/raw/boats.csv"
+    file_format: parquet
+
+raw_car_data:
+-    type: "${datasets.csv}"
++    type: "${_datasets.csv}"
+-    filepath: "s3://${bucket_name}/data/${key_prefix}/raw/cars.csv"
++    filepath: "s3://${_bucket_name}/data/${_key_prefix}/raw/cars.csv"
+```
+
+#### Providing default values for templates via `oc.select`
+To provide a default for any template values you have to use [the omegaconf `oc.select` resolver](https://omegaconf.readthedocs.io/en/latest/custom_resolvers.html#oc-select).
+
+```diff
+boats:
+  users:
+    - fred
+-    - "${write_only_user|ron}"
++    - "${oc.select:write_only_user,ron}"
+```
+
+### 7. Globals
+If you want to share variables across configuration types, e.g. parameters and catalog, and environments you need to use [the custom globals resolver with the `OmegaConfigLoader`](advanced_configuration.md#how-to-use-global-variables-with-the-omegaconfigloader).
+The `OmegaConfigLoader` requires global values to be provided in a `globals.yml` file. The following section explains the differences between using globals with `TemplatedConfigLoader` and the `OmegaConfigLoader`.
+
+Let's assume your project contains a `conf/base/globals.yml` file with the following contents:
+
+```yaml
+bucket_name: "my_s3_bucket"
+key_prefix: "my/key/prefix/"
+
+datasets:
+    csv: "pandas.CSVDataSet"
+    spark: "spark.SparkDataSet"
+
+folders:
+    raw: "01_raw"
+    int: "02_intermediate"
+    pri: "03_primary"
+    fea: "04_feature"
+```
+
+You no longer need to set `CONFIG_LOADER_ARGS` variable in [`src/<package_name>/settings.py`](../kedro_project_setup/settings.md) to find this `globals.yml` file, because the
+`OmegaConfigLoader` is configured to pick up files named `globals.yml` by default.
```

Comment:
Should we add a link in case they name it something other than `globals.yml`?

Reply:
What's the replacement for `globals_dict`?

Reply:
There's no replacement. That's why I wrote "The `OmegaConfigLoader` requires global values to be provided in a `globals.yml` file.", but I can explicitly mention that the `globals_dict` is not supported with OCL for clarity.

Reply:
"Should we add a link in case they name it something other than globals.yml?" @noklam what should we link to?

Reply:
@merelcht How about https://docs.kedro.org/en/latest/configuration/advanced_configuration.html#how-to-use-global-variables-with-the-omegaconfigloader?

Btw I think we should bubble up OmegaConfigLoader docs since we are trying to make it default. The hierarchy is also broken there is nothing inside OmegaConfigLoader. Cc @stichbury @ankatiyar 
![image](https://github.com/kedro-org/kedro/assets/18221871/217d0944-d20c-4408-a617-0fd1ea7951ab)

Reply:
> The hierarchy is also broken there is nothing inside OmegaConfigLoader.

I don't think it's broken, we don't have any sub-sections under `OmegaConfigLoader` because those are all under "Advanced Kedro configuration". We should probably clean this up when all docs on `ConfigLoader` and `TemplatedConfigLoader` are removed as part of: https://github.com/kedro-org/kedro/issues/2692

Reply:
@noklam We will be doing some of the moving around sections in the docs for https://github.com/kedro-org/kedro/issues/2900 after the "make starters use `OmegaConfigLoader` by default" PRs are merged. (@lrcouto)